### PR TITLE
rework callbacks

### PR DESCRIPTION
**Pull request by benjamin-work**
I believe that especially if one looks at the changes in `net.py`, the new changes should result in cleaner user code.

### CODE REVIEW

File: inferno/callbacks.py [hunk 1]
```diff
@@ -140,46 +153,58 @@ class BestLoss(Callback):
...
                 loss = history[-1, key]
                 yield key, sign, loss
             except KeyError:
                 pass
+        for key, sign in self.key_signs.items():
+            loss = history[-1, key]
+            yield key, sign, loss
 
     def on_epoch_end(self, net, **kwargs):
+        sl = slice(-1, None), list(self.key_signs)
```
Comment (ottonemo) [inline inferno/callbacks.py]: Why  not use `np.s_` here as well?

File: inferno/callbacks.py [hunk 2]
```diff
@@ -267,6 +292,10 @@ class PrintLog(Callback):
     `BestLoss` callback takes care of creating those entries, which is
     why `PrintLog` works best in conjunction with that callback.
 
+    *Note*: `PrintLog` will not result in good outputs if the number
```
Comment (ottonemo) [inline inferno/callbacks.py]: s/result in/yield/

File: inferno/callbacks.py (hunk 1 above)
Comment (benjamin-work) [inline inferno/callbacks.py]: :+1:

File: inferno/callbacks.py (hunk 2 above)
Comment (benjamin-work) [inline inferno/callbacks.py]: Not sure here. In Python docstrings, I prefer to use "yield" specifically to mean that a function `yield`s things, which is not the case here. Would the change result in more clarity?

Comment (ottonemo) [inline inferno/callbacks.py]: Fair point. It is a minor issue at best so feel free to ignore. Pretty sure that people will understand what is meant.

**Pull request merged**